Detectors/EMCAL/base/include/EMCALBase/Geometry.h

```diff
@@ -658,11 +661,42 @@ class Geometry
   Float_t mZLength;        ///< Total length in z direction
   Float_t mSampling;       ///< Sampling factor
 
+  // Members from the EMCGeometry class
+  Float_t mECPbRadThickness; ///< cm, Thickness of the Pb radiators
+  Float_t mECScintThick;     ///< cm, Thickness of the scintillators
+  Int_t mNECLayers;          ///< number of scintillator layers
+
+  // Shish-kebab option - 23-aug-04 by PAI; COMPACT, TWIST, TRD1 and TRD2
+  Int_t mNumberOfSuperModules; ///< default is 12 = 6 * 2
+
+  /// geometry structure
+  Int_t* mEMCSMSystem; //[mNumberOfSuperModules]
```

Comment:
Not sure. The container should still have a fixed size and not be expandable, only that the fixed size depends on the geometry version. For the moment I would rather prefer this or TArray from ROOT. Maybe in the next development step I will try to decouple the various geometry setups as child classes inheriting from Geometry - this would then allow to define constants i.e. for the number of supermodules.